src/cc/arduino/plugins/wifi101/flashers/java/NinaFlasher.java

```diff
@@ -0,0 +1,338 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As a special exception, you may use this file as part of a free software
+ * library without restriction.  Specifically, if other files instantiate
+ * templates or use macros or inline functions from this file, or you compile
+ * this file and link it with other files to produce an executable, this
+ * file does not by itself cause the resulting executable to be covered by
+ * the GNU General Public License.  This exception does not however
+ * invalidate any other reasons why the executable file might be covered by
+ * the GNU General Public License.
+ */
+package cc.arduino.plugins.wifi101.flashers.java;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.*;
+import java.util.List;
+import java.io.ByteArrayOutputStream;
+import java.io.*;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import org.apache.commons.lang3.StringUtils;
+import java.security.MessageDigest;
+
+import cc.arduino.plugins.wifi101.certs.WiFi101Certificate;
+import cc.arduino.plugins.wifi101.certs.WiFi101CertificateBundle;
+
+import cc.arduino.plugins.wifi101.flashers.Flasher;
+import javax.swing.JProgressBar;
+
+public class NinaFlasher implements Flasher {
+  public String boardName;
+  public String version;
+  public File file;
+  public File Compatiblefile;
+  public JProgressBar progressBar;
+  public String name;
+  public List<String> compatibleboard;
+  public byte[] md5Checksum;
+
+  public NinaFlasher(String _name, String _version, String _filename) {
+    boardName = _name;
+    compatibleboard =  new ArrayList<String>();
+    version = _version;
+    file = null;
+    name = "WifiNINA";
+
+    try {
+      String jarPath = Flasher.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath();
+      File jarFolder = new File(jarPath).getParentFile();
+      file = new File(jarFolder, _filename);
+      md5Checksum = getMD5Checksum(file.getPath());
+    }
+    catch (URISyntaxException e) {
+      e.printStackTrace();
+    }
+    catch (Exception e) {
+      e.printStackTrace();
+    }
+    try {
+      String line = null;
+      String jarPath = Flasher.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath();
+      File jarFolder = new File(jarPath).getParentFile();
+      Compatiblefile = new File(jarFolder, "firmwares/Config.ini");
```

Comment:
Mmmmh, a CSV style file would be **a bit** better, instead than parsing the strings

Reply:
Previously the list was hard coded in the .java file and reason to move away from this? Things seemed much simpler.